gnuradio-runtime/lib/top_block_impl.cc

```diff
@@ -129,6 +132,21 @@ namespace gr {
   void
   top_block_impl::wait()
   {
+    do {
```

Comment:
It waits for jobs to complete. Jobs complete after call of stop(). The checks whatever the block is locked, if not then exit, if is locked, wait until is unlocked and do both checks again. The point is to prevent exiting from wait() whet the block is locked but stoped. Until now this behaviour was simulated trought consectuive calls of stop() wait() start() wich does not given to other calls of wait() enough time to rally exit.